File: frontend-web/src/services/noteTemplatesService.js
/**
 * Service pour gérer les templates de notes
 */
import apiClient from './api';

export const noteTemplatesService = {
  /**
   * Créer un template
   */
  async createTemplate(name, description, content, category = 'general', isPublic = false) {
    const response = await apiClient.post('/note-templates', {
      name,
      description,
      content,
      category,
      is_public: isPublic,
    });
    return response.data;
  },

  /**
   * Lister les templates
   */
  async listTemplates(category = null) {
    const response = await apiClient.get('/note-templates', {
      params: category ? { category } : {},
    });
    return response.data;
  },

  /**
   * Créer une note depuis un template
   */
  async createNoteFromTemplate(templateId, title = null, folderId = null) {
    const response = await apiClient.post(`/note-templates/${templateId}/create-note`, {
      title,
      folder_id: folderId,
    });
    return response.data;
  },

  /**
   * Mettre à jour un template
   */
  async updateTemplate(templateId, updates) {
    const response = await apiClient.patch(`/note-templates/${templateId}`, updates);
    return response.data;
  },

  /**
   * Supprimer un template
   */
  async deleteTemplate(templateId) {
    const response = await apiClient.delete(`/note-templates/${templateId}`);
    return response.data;
  },
};

File: mobile-app/lib/services/note_templates_service.dart
import '../models/note_template.dart';
import '../models/note.dart';
import 'api_service.dart';

class NoteTemplatesService {
  final ApiService _apiService = ApiService();

  /// Lister les templates
  Future<List<NoteTemplate>> listTemplates({String? category}) async {
    try {
      final response = await _apiService.get(
        '/note-templates',
        queryParameters: category != null ? {'category': category} : null,
      );
      final templates = (response.data['templates'] as List)
          .map((e) => NoteTemplate.fromJson(e))
          .toList();
      return templates;
    } catch (e) {
      throw Exception('Erreur lors du chargement des templates: $e');
    }
  }

  /// Créer une note depuis un template
  Future<Note> createNoteFromTemplate(String templateId, {String? title, String? folderId}) async {
    try {
      final response = await _apiService.post(
        '/note-templates/$templateId/create-note',
        data: {
          if (title != null) 'title': title,
          if (folderId != null) 'folder_id': folderId,
        },
      );
      return Note.fromJson(response.data['note']);
    } catch (e) {
      throw Exception('Erreur lors de la création depuis le template: $e');
    }
  }
}




